fili-core/src/main/java/com/yahoo/bard/webservice/async/workflows/DefaultAsynchronousWorkflowsBuilder.java

```diff
@@ -204,8 +206,8 @@ public AsynchronousWorkflows buildAsynchronousWorkflows(
         // The job status should not be updated until both the PreResponse has been stored, and the storage of the
         // original Job ticket has been attempted.
         return preResponseEmitter
-                .delay(ignored -> preResponseStoredNotification)
-                .delay(ignored -> jobRowStoredNotification)
+                .zipWith(jobRowStoredNotification, (preResponse, ignored) -> preResponse)
+                .zipWith(preResponseStoredNotification, (preResponse, ignored) -> preResponse)
```

Comment:
It is, but I don't really know of a better way. Not if we want to release it iff the two observables emit items.